components/google_sheets/sources/common/http-based/drive.mjs

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable pipedream/required-properties-key, pipedream/required-properties-name,
+  pipedream/required-properties-version, pipedream/required-properties-description,
+  pipedream/required-properties-type */
+
+import { v4 as uuid } from "uuid";
+
+import base from "./base.mjs";
+
+/**
+ * This source watches for all changes to a shared drive, processing only
+ * changes to the chosen file.
+ */
+export default {
+  ...base,
+  props: {
+    ...base.props,
+    watchedDrive: {
+      propDefinition: [
+        base.props.googleSheets,
+        "watchedDrive",
+      ],
+      description: "Defaults to My Drive. To select a [Shared Drive](https://support.google.com/a/users/answer/9310351) instead, select it from this list.",
+      optional: false,
+    },
+  },
+  hooks: {
+    ...base.hooks,
+    /**
+     * Called when a component is created or updated. Handles all the logic
+     * for starting and stopping watch notifications tied to the desired file.
+     */
+    async activate() {
+      const channelID = this._getChannelID() || uuid();
+
+      const {
+        startPageToken,
+        expiration,
+        resourceId,
+      } = await this.googleSheets.activateHook(
+        channelID,
+        this.http.endpoint,
+        this.getDriveId(),
+      );
+
+      // Save metadata on the subscription so we can stop / renew later
+      // Subscriptions are tied to Google's resourceID, "an opaque value that
+      // identifies the watched resource". This value is included in request headers
+      this._setSubscription({
+        resourceId,
+        expiration,
+      });
+      this._setChannelID(channelID);
+      this._setPageToken(startPageToken);
+
+      await this.takeSheetSnapshot();
+    },
+  },
+  methods: {
+    ...base.methods,
+    _getPageToken() {
+      return this.db.get("pageToken");
+    },
+    _setPageToken(pageToken) {
+      this.db.set("pageToken", pageToken);
+    },
+    async getModifiedSheet(pageToken, driveId, sheetID) {
```

Comment:
This would be called from the `activate()` hook in base.mjs
```suggestion
    },
    async activateHook(channelId) {
      return this.googleSheets.activateHook(
        this.http.endpoint,
        this.getDriveId(),
      );
    },
    async getModifiedSheet(pageToken, driveId, sheetID) {
```